### PR TITLE
[Feature]: Main Task OS Sanity Check

### PR DESCRIPTION
**Pull request by brolan-juul**
Added ability to include `os_sanity` check with `main task`.
Developers will have to create their os_sanity check-in in their application.

### CODE REVIEW

File: kernel/os/src/os.c [hunk 1]
```diff
@@ -45,8 +45,14 @@ OS_TASK_STACK_DEFINE(g_idle_task_stack, OS_IDLE_STACK_SIZE);
...
 
-static struct os_task os_main_task;
-OS_TASK_STACK_DEFINE(os_main_stack, OS_MAIN_STACK_SIZE);
+struct os_task g_os_main_task;
+OS_TASK_STACK_DEFINE(g_os_main_stack, OS_MAIN_STACK_SIZE);
+
+/*
+ * Double the interval timer to allow proper timer check-in.
+ */
+#define OS_MAIN_TASK_TIMER_TICKS \
```
Comment (andrzej-kaczmarek) [inline kernel/os/src/os.c]: this should be called smth like `OS_MAIN_TASK_SANITY_ITVL_TICKS`

File: kernel/os/src/os.c [hunk 2]
```diff
@@ -45,8 +45,14 @@ OS_TASK_STACK_DEFINE(g_idle_task_stack, OS_IDLE_STACK_SIZE);
 
 uint32_t g_os_idle_ctr;
 
-static struct os_task os_main_task;
-OS_TASK_STACK_DEFINE(os_main_stack, OS_MAIN_STACK_SIZE);
+struct os_task g_os_main_task;
+OS_TASK_STACK_DEFINE(g_os_main_stack, OS_MAIN_STACK_SIZE);
+
+/*
+ * Double the interval timer to allow proper timer check-in.
```
Comment (andrzej-kaczmarek) [inline kernel/os/src/os.c]: I looked quickly through os_sanity code and honestly have no idea why need to double the interval

File: kernel/os/src/os.c (hunk 1 above)
Comment (brolan-juul) [reply kernel/os/src/os.c]: Agreed. That would be a more descriptive name.

File: kernel/os/src/os.c (hunk 2 above)
Comment (brolan-juul) [reply kernel/os/src/os.c]: Removing then need for this. Thanks!

File: kernel/os/src/os.c [hunk 3]
```diff
@@ -235,8 +232,8 @@ os_init(int (*main_fn)(int argc, char **arg))
     if (main_fn) {
         err = os_task_init(&g_os_main_task, "main", os_main, main_fn,
                    OS_MAIN_TASK_PRIO,
-                   (OS_MAIN_TASK_TIMER_TICKS == 0) ? \
-                        OS_WAIT_FOREVER : OS_MAIN_TASK_TIMER_TICKS,
+                   (OS_MAIN_TASK_SANITY_ITVL_TICKS == 0) ? \
+                        OS_WAIT_FOREVER : OS_MAIN_TASK_SANITY_ITVL_TICKS,
```
Comment (wes3) [inline kernel/os/src/os.c]: No need for \ here btw. Should work without it. Looks fine otherwise.

Comment (brolan-juul) [reply kernel/os/src/os.c]: I spaced it to two lines to keep it under 80 characters per line tho I'd have to admit, I haven't read the style guide for this project. I'll remove the `\`.

**Pull request merged**